### PR TITLE
Introduce checkbox property type

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR introduces a checkbox property type. The `Switch` component is used to render the checkbox's value, though this is likely not the intended solution.

The discussion in the linked ticket https://github.com/mattermost/focalboard/issues/39 suggests to look at the [implementation of the checkbox content type](https://github.com/mattermost/focalboard/pull/146) for reference, but I wasn't able to connect `widgets/propertyValueElement.tsx` with `content/checkboxElement.tsx`. I think I'm missing a fundamental understanding of how the checkbox element fits into the property value code.

Also, all property values seem to be strings at the moment. What `string <-> boolean` system should be used here? It seems that a card's property values can either a non-empty string, or an undefined value. The PR currently uses `'true' === true` and `undefined === false`, and never sets the value to `'false'`.

Some webapp development tooling was changed as well.

#### Ticket Link

Fixes https://github.com/mattermost/focalboard/issues/39

### CODE REVIEW

File: webapp/src/components/propertyValueElement.tsx [hunk 1]
```diff
@@ -115,6 +116,19 @@ const PropertyValueElement = (props:Props): JSX.Element => {
         )
     }
 
+    if (propertyTemplate.type === 'checkbox') {
+        return (
+            <Switch
+                isOn={Boolean(propertyValue)}
+                onChanged={(newBool) => {
+                    const newValue = newBool ? 'true' : '';
```
Comment (chenilim) [inline webapp/src/components/propertyValueElement.tsx:124]: There's probably a longer-term architecture question to think about here (fyi @jespino and @sbishel). 

It's possible to overload changePropertyValue to accept boolean values (like we do for arrays in the [multi-select PR](https://github.com/mattermost/focalboard/pull/415/files)). That would be cleaner, but I can't think of any problems with this approach either (which is simpler). One minor thing is maybe to store true as '1' (to make conversion to number easier), but that's 0/5.

As a side-note, we should probably test all the conversions between property types (i.e. by changing the type of an existing property with values, fyi @ogi-m) and flag any issues.

**Comment by harshilsharma63** [comment]
@chenilim @sbishel this is a decision we have to take whether we want to use the switch style of the regular checkbox style for this.

Thoughts?

**Comment by chenilim** [comment]
We should make this a "standard" checkbox style, but I'm ok taking that as a subsequent change.

**Comment by harshilsharma63** [comment]
@mickmister the changes look good and works nicely! Thanks a lot!

Can you fix the lint issues though?

File: webapp/src/components/propertyValueElement.tsx (hunk 1 above)
Comment (jespino) [reply webapp/src/components/propertyValueElement.tsx:124]: I think I prefer to store the right value when it is possible, in this case, a boolean, is a boolean, moving into this "true"/"false" approach is something that is hard to change later.

File: webapp/webpack.dev.js [hunk 2]
```diff
@@ -16,5 +16,9 @@ const config = merge.merge(commonConfig, {
 
 module.exports = [
     merge.merge(config, {
+        devtool: 'source-map',
```
Comment (jespino) [inline webapp/webpack.dev.js:19]: nice

**Review by harshilsharma63 — APPROVED** [review]
LGTM

**Pull request merged**